backtester: Investigate TestStart race

Spotted here: https://github.com/thrasher-corp/gocryptotrader/actions/runs/16561869416/job/46833027347#step:11:92

```
==================
WARNING: DATA RACE
Write at 0x00c0006fe748 by goroutine 52:
  github.com/thrasher-corp/gocryptotrader/backtester/engine.(*dataChecker).Start()
      /home/runner/work/gocryptotrader/gocryptotrader/backtester/engine/live.go:79 +0x1a4
  github.com/thrasher-corp/gocryptotrader/backtester/engine.TestStart()
      /home/runner/work/gocryptotrader/gocryptotrader/backtester/engine/live_test.go:77 +0x151
  testing.tRunner()
      /opt/hostedtoolcache/go/1.24.5/x64/src/testing/testing.go:1792 +0x225
  testing.(*T).Run.gowrap1()
      /opt/hostedtoolcache/go/1.24.5/x64/src/testing/testing.go:1851 +0x44

Previous read at 0x00c0006fe748 by goroutine 126:
  github.com/thrasher-corp/gocryptotrader/backtester/engine.(*dataChecker).SignalStopFromError()
      /home/runner/work/gocryptotrader/gocryptotrader/backtester/engine/live.go:122 +0x199
  github.com/thrasher-corp/gocryptotrader/backtester/engine.(*dataChecker).Start.func1()
      /home/runner/work/gocryptotrader/gocryptotrader/backtester/engine/live.go:83 +0x7c

Goroutine 52 (running) created at:
  testing.(*T).Run()
      /opt/hostedtoolcache/go/1.24.5/x64/src/testing/testing.go:1851 +0x8f2
  testing.runTests.func1()
      /opt/hostedtoolcache/go/1.24.5/x64/src/testing/testing.go:2279 +0x85
  testing.tRunner()
      /opt/hostedtoolcache/go/1.24.5/x64/src/testing/testing.go:1792 +0x225
  testing.runTests()
      /opt/hostedtoolcache/go/1.24.5/x64/src/testing/testing.go:2277 +0x96c
  testing.(*M).Run()
      /opt/hostedtoolcache/go/1.24.5/x64/src/testing/testing.go:2142 +0xeea
  main.main()
      _testmain.go:179 +0x164

Goroutine 126 (running) created at:
  github.com/thrasher-corp/gocryptotrader/backtester/engine.(*dataChecker).Start()
      /home/runner/work/gocryptotrader/gocryptotrader/backtester/engine/live.go:80 +0x244
  github.com/thrasher-corp/gocryptotrader/backtester/engine.TestStart()
      /home/runner/work/gocryptotrader/gocryptotrader/backtester/engine/live_test.go:71 +0xe5
  testing.tRunner()
      /opt/hostedtoolcache/go/1.24.5/x64/src/testing/testing.go:1792 +0x225
  testing.(*T).Run.gowrap1()
      /opt/hostedtoolcache/go/1.24.5/x64/src/testing/testing.go:1851 +0x44
==================
--- FAIL: TestStart (0.00s)
    live_test.go:78: 
        	Error Trace:	/home/runner/work/gocryptotrader/gocryptotrader/backtester/engine/live_test.go:78
        	Error:      	Target error should be in err chain:
        	            	expected: "subsystem already started"
        	            	in chain: 
        	Test:       	TestStart
    testing.go:1490: race detected during execution of test
--- FAIL: TestFetchLatestData (0.03s)
    testing.go:1490: race detected during execution of test
--- FAIL: TestSetupFromConfig (21.28s)
    testing.go:1490: race detected during execution of test
--- FAIL: TestExecuteStrategyFromConfig (40.46s)
    testing.go:1490: race detected during execution of test
FAIL
coverage: 73.4% of statements
FAIL	github.com/thrasher-corp/gocryptotrader/backtester/engine	41.027s
```

--- a/backtester/engine/live_test.go
+++ b/backtester/engine/live_test.go
@@ -2,7 +2,6 @@ package engine
 
 import (
 	"sync"
-	"sync/atomic"
 	"testing"
 	"time"
 
@@ -65,20 +64,23 @@ func TestSetupLiveDataHandler(t *testing.T) {
 
 func TestStart(t *testing.T) {
 	t.Parallel()
-	dc := &dataChecker{
+	dc1 := &dataChecker{
 		shutdown: make(chan bool),
 	}
-	err := dc.Start()
+	err := dc1.Start()
 	assert.NoError(t, err)
 
-	close(dc.shutdown)
-	dc.wg.Wait()
-	atomic.CompareAndSwapUint32(&dc.started, 0, 1)
-	err = dc.Start()
+	close(dc1.shutdown)
+	dc1.wg.Wait()
+
+	dc2 := &dataChecker{
+		started: 1,
+	}
+	err = dc2.Start()
 	assert.ErrorIs(t, err, engine.ErrSubSystemAlreadyStarted)
 
-	var dh *dataChecker
-	err = dh.Start()
+	var dc3 *dataChecker
+	err = dc3.Start()
 	assert.ErrorIs(t, err, gctcommon.ErrNilPointer)
 }
 

--- a/backtester/engine/live_test.go
+++ b/backtester/engine/live_test.go
@@ -64,33 +64,25 @@ func TestSetupLiveDataHandler(t *testing.T) {
 
 func TestStart(t *testing.T) {
 	t.Parallel()
-	t.Run("successful start", func(t *testing.T) {
-		t.Parallel()
-		dc := &dataChecker{
-			shutdown: make(chan bool),
-		}
-		err := dc.Start()
-		assert.NoError(t, err)
 
-		close(dc.shutdown)
-		dc.wg.Wait()
-	})
+	var dc *dataChecker
+	err := dc.Start()
+	assert.ErrorIs(t, err, gctcommon.ErrNilPointer)
 
-	t.Run("already started", func(t *testing.T) {
-		t.Parallel()
-		dc := &dataChecker{
-			started: 1,
-		}
-		err := dc.Start()
-		assert.ErrorIs(t, err, engine.ErrSubSystemAlreadyStarted)
-	})
+	dc = &dataChecker{
+		shutdown: make(chan bool),
+	}
+	err = dc.Start()
+	require.NoError(t, err)
 
-	t.Run("nil pointer", func(t *testing.T) {
-		t.Parallel()
-		var dc *dataChecker
-		err := dc.Start()
-		assert.ErrorIs(t, err, gctcommon.ErrNilPointer)
-	})
+	close(dc.shutdown)
+	dc.wg.Wait()
+
+	dc = &dataChecker{
+		started: 1,
+	}
+	err = dc.Start()
+	assert.ErrorIs(t, err, engine.ErrSubSystemAlreadyStarted)
 }
 
 func TestDataCheckerIsRunning(t *testing.T) {

--- a/backtester/engine/live_test.go
+++ b/backtester/engine/live_test.go
@@ -64,24 +64,33 @@ func TestSetupLiveDataHandler(t *testing.T) {
 
 func TestStart(t *testing.T) {
 	t.Parallel()
-	dc1 := &dataChecker{
-		shutdown: make(chan bool),
-	}
-	err := dc1.Start()
-	assert.NoError(t, err)
+	t.Run("successful start", func(t *testing.T) {
+		t.Parallel()
+		dc := &dataChecker{
+			shutdown: make(chan bool),
+		}
+		err := dc.Start()
+		assert.NoError(t, err)
 
-	close(dc1.shutdown)
-	dc1.wg.Wait()
+		close(dc.shutdown)
+		dc.wg.Wait()
+	})
 
-	dc2 := &dataChecker{
-		started: 1,
-	}
-	err = dc2.Start()
-	assert.ErrorIs(t, err, engine.ErrSubSystemAlreadyStarted)
+	t.Run("already started", func(t *testing.T) {
+		t.Parallel()
+		dc := &dataChecker{
+			started: 1,
+		}
+		err := dc.Start()
+		assert.ErrorIs(t, err, engine.ErrSubSystemAlreadyStarted)
+	})
 
-	var dc3 *dataChecker
-	err = dc3.Start()
-	assert.ErrorIs(t, err, gctcommon.ErrNilPointer)
+	t.Run("nil pointer", func(t *testing.T) {
+		t.Parallel()
+		var dc *dataChecker
+		err := dc.Start()
+		assert.ErrorIs(t, err, gctcommon.ErrNilPointer)
+	})
 }
 
 func TestDataCheckerIsRunning(t *testing.T) {

